internal/policy/policy.go

```diff
@@ -629,6 +634,53 @@ func (s *State) HasTargetsRole(roleName string) bool {
 	return ok
 }
 
+// ListRules returns a list of all the rules as an array of the delegations in a pre order traversal of the
+// delegation tree, with the depth of each delegation.
+func ListRules(ctx context.Context, repo *git.Repository) ([]*DelegationWithDepth, error) {
+	state, err := LoadCurrentState(ctx, repo)
+	if err != nil {
+		return nil, err
+	}
+
+	topLevelTargetsMetadata, err := state.GetTargetsMetadata(TargetsRoleName)
+	if err != nil {
+		return nil, err
+	}
+
+	delegationsToSearch := []*DelegationWithDepth{}
+	allDelegations := []*DelegationWithDepth{}
+
+	for _, topLevelDelegation := range topLevelTargetsMetadata.Delegations.Roles {
+		delegationsToSearch = append(delegationsToSearch, &DelegationWithDepth{Delegation: topLevelDelegation, Depth: 0})
+	}
+
+	for len(delegationsToSearch) > 0 {
+		currentDelegation := delegationsToSearch[0]
+		delegationsToSearch = delegationsToSearch[1:]
+
+		// allDelegations will be the returned list of all the delegations in pre-order traversal, no delegations will be popped off
+		allDelegations = append(allDelegations, currentDelegation)
+		if state.HasTargetsRole(currentDelegation.Delegation.Name) {
+			currentMetadata, err := state.GetTargetsMetadata(currentDelegation.Delegation.Name)
+			if err != nil {
+				return nil, err
+			}
+
+			delegationsToSearch = append([]*DelegationWithDepth{{Delegation: currentMetadata.Delegations.Roles[len(currentMetadata.Delegations.Roles)-1], Depth: currentDelegation.Depth + 1}}, delegationsToSearch...)
```

Comment:
Can you clarify this statement along with line 676?

Reply:
This line is now gone, since we do not need the additional allow rules